storyblok/field-plugin
# EXT-1318: Disable Shields

## What?

When opening the container for the first time, the user is notified that ad-blockers and shields must be disabled.

Note that this is not based on designs from the design team, so it will require an update later on.

## Why?

When the container is not served from `localhost`, Brave will block the iframe because it points to (another) `localhost` origin.

In the Storyblok app, this is a reliable source of frustration for first-time developers of field plugins, because it's not so obvious why the result does not show up.

I have tried to prevent the iframe from being blocked in the first place, but it doesn't seem possible unless you control the domain. For example, it is possible to serve without diabling shields if serving over a secure tunnel.

I have also tried to detect whether the iframe was blocked via JavaScript, but this doesn't seem possible either.

## How to test? (optional)

Open the container in the Vercel Preview. (Do not serve locally).

Ensure that the shields are enabled. This should block the iframe.

The notification will show up. When you close it, it will never re-open.

Disable the shields. This should allow the iframe to load.

<img width="1512" alt="image" src="https://user-images.githubusercontent.com/14206504/231396503-6d59da46-db51-4ed4-92ff-bc409b750371.png">
